File: frontend/lib/features/conversational_submission/domain/entities/dealer_result.dart
import 'package:equatable/equatable.dart';

/// A dealer result from state-based dealer typeahead search.
class DealerResult extends Equatable {
  final String dealerCode;
  final String dealerName;
  final String? city;
  final String state;

  const DealerResult({
    required this.dealerCode,
    required this.dealerName,
    this.city,
    required this.state,
  });

  @override
  List<Object?> get props => [dealerCode, dealerName, city, state];
}

File: backend/src/BajajDocumentProcessing.Application/DTOs/Conversation/DealerResult.cs
using System.Text.Json.Serialization;

namespace BajajDocumentProcessing.Application.DTOs.Conversation;

/// <summary>
/// Dealer typeahead result from the StateMapping table
/// </summary>
public class DealerResult
{
    /// <summary>
    /// Dealer code identifier
    /// </summary>
    [JsonPropertyName("dealerCode")]
    public required string DealerCode { get; init; }

    /// <summary>
    /// Dealer name
    /// </summary>
    [JsonPropertyName("dealerName")]
    public required string DealerName { get; init; }

    /// <summary>
    /// City where the dealer is located
    /// </summary>
    [JsonPropertyName("city")]
    public required string City { get; init; }

    /// <summary>
    /// State where the dealer is located
    /// </summary>
    [JsonPropertyName("state")]
    public required string State { get; init; }
}
